src/edge/edgeManager.ts

```diff
@@ -177,6 +177,86 @@ export class EdgeManager {
         }
     }
 
+    public async loadWebView(outputChannel: vscode.OutputChannel) {
+      const panel = vscode.window.createWebviewPanel(Constants.galleryPanelViewType, Constants.galleryPanelViewTitle, vscode.ViewColumn.One, {
+        enableScripts: true,
+        retainContextWhenHidden: true,
+      });
+
+      const srcPath: string = this.context.asAbsolutePath(path.join(Constants.assetsFolder, Constants.galleryAssetsFolder, Constants.galleryIndexHtmlName));
+      panel.webview.html = await this.getWebViewContent(srcPath);
+
+      // Handle messages from the webview
+      panel.webview.onDidReceiveMessage(async (message) => {
+        switch (message.command) {
+          case "openSample":
+            if (message.name && message.url) {
+              await this.initializeSample(message.name, message.url, message.platform, outputChannel);
+            }
+            break;
+          case "openLink":
+            if (message.url) {
+              TelemetryClient.sendEvent(Constants.openSampleUrlEvent, {Url: message.url});
+              await vscode.commands.executeCommand("vscode.open", vscode.Uri.parse(message.url));
+            }
+            break;
+        }
+      }, undefined, this.context.subscriptions);
+    }
+
+    public async getWebViewContent(templatePath: string): Promise<string> {
+      const dirPath = path.dirname(templatePath);
+      let html = await fse.readFile(templatePath, "utf-8");
+
+      html = html.replace(/(<link.*?\shref="|<script.*?\ssrc="|<img.*?\ssrc=")(.+?)"/g, (m, $1, $2) => {
+          return $1 + vscode.Uri.file(path.join(dirPath, $2)).with({ scheme: "vscode-resource" }).toString(true) + "\"";
+      });
+      return html;
+    }
+
+    public async initializeSample(name: string, url: string, platform: string, channel: vscode.OutputChannel) {
+      try {
+        const parentPath = await this.getSolutionParentFolder();
+
+        if (parentPath === undefined) {
+          throw new UserCancelledError();

```

Comment:
is this error handled by outer side?

Reply:
Nope, however, i think we do not need to throw error in the initializeSample function.
![image](https://user-images.githubusercontent.com/5545529/51893526-1a283580-23e0-11e9-9fee-a08053a1f057.png)